hpilo.py

```diff
@@ -427,7 +429,13 @@ def close(self):
             raise IloCommunicationError("Unable to resolve %s" % self.hostname)
 
         try:
-            return ssl.wrap_socket(sock, ssl_version=self.ssl_version)
+            cert_reqs=ssl.CERT_NONE
+            if self.ca_file:
+              cert_reqs=ssl.CERT_REQUIRED
+            sslContext = ssl.wrap_socket(sock, ssl_version=self.ssl_version, cert_reqs=cert_reqs, ca_certs=self.ca_file)
+            if self.ca_file and not self.ignore_hostname:
+              ssl.match_hostname(sslContext.getpeercert(), self.hostname)
```

Comment:
Same here, we need something (a match_hostname that does nothing perhaps) in the backwards compatibility shim.